pkg/apis/core/validation/validation.go

```diff
@@ -3018,6 +3018,10 @@ func ValidatePodSpec(spec *core.PodSpec, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, ValidateRuntimeClassName(*spec.RuntimeClassName, fldPath.Child("runtimeClassName"))...)
 	}
 
+	if spec.Overhead != nil {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("overhead"), "may only be set during admission"))
+	}
```

Comment:
Hmm, I'm not sure whether validation runs before or after admission... my guess would be after since we would want to validate admission mutations...

Also, this should be ignored if the PodOverhead feature is disabled.

I think it might make more sense to just enforce this in the admission controller. That way we can also ignore the error if it is set to the same value as the RuntimeClass.

Reply:
Makes sense.  Addressed.